apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/content/subsection/SubSectionCreator.java

```diff
@@ -43,26 +50,30 @@ public CreateSubSectionResponse createSubSection(long pageId, Long userId, Creat
             throw new AddSubSectionException("Failed to add SubSection");
         }
     }
-
-    private static final long TAB = 1010L;
-    private static final long SECTION = 1015L;
-    private static final long SUBSECTION = 1016L;
-
     public DeleteSubSectionResponse deleteSubSection(DeleteSubSectionRequest request) {
         try {
-            log.info("Deleting sub section");
+            log.info("Deleting Sub Section");
             Integer orderNbr = waUiMetaDataRepository.getOrderNumber(request.subSectionId());
-            Long pageNumber = waUiMetaDataRepository.findPageNumber(request.subSectionId());
-            Long nbsComponentUid = waUiMetaDataRepository.findNextNbsUiComponentUid(orderNbr+1, pageNumber);
-            if(nbsComponentUid == TAB || nbsComponentUid == SECTION || nbsComponentUid == null || nbsComponentUid == SUBSECTION) {
+            Long pageNumber = waUiMetaDataRepository.findPageNumber(request.subSectionId()).getId();
+            Optional<Long> nbsComponentUidOptional =
+                    waUiMetaDataRepository.findNextNbsUiComponentUid(orderNbr+1, pageNumber);
+            if (nbsComponentUidOptional.isPresent()) {
+                Long nbsComponentUid = nbsComponentUidOptional.get();
+                if (nbsComponentUid == TAB ||nbsComponentUid == SECTION
+                        ||nbsComponentUid == SUBSECTION || nbsComponentUid == null) {
+                    waUiMetaDataRepository.deleteById(request.subSectionId());
+                    waUiMetaDataRepository.updateOrderNumberByDecreasing(orderNbr, request.subSectionId());
+                    return new DeleteSubSectionResponse(request.subSectionId(), DELETE_MESSAGE);
+                } else {
+                    throw new DeleteSubSectionException("Conditions not satisfied");
+                }
+            } else {
                 waUiMetaDataRepository.deleteById(request.subSectionId());
                 waUiMetaDataRepository.updateOrderNumberByDecreasing(orderNbr, request.subSectionId());
-                return new DeleteSubSectionResponse(request.subSectionId(), "Sub Section Deleted Successfully");
-            } else {
-                throw new DeleteSubSectionException("Conditions not satisfied");
+                return new DeleteSubSectionResponse(request.subSectionId(), DELETE_MESSAGE);
             }
         } catch(Exception exception) {
-            throw new DeleteSubSectionException("Delete Sub Section exception");
+            throw new DeleteSubSectionException(exception.toString());
```

Comment:
We don't want to pass the message of the real exception. These could contain stack traces.